crates/viewer/re_space_view_spatial/src/mesh_loader.rs

```diff
@@ -52,11 +52,12 @@ impl LoadedMesh {
             _ => anyhow::bail!("{media_type} files are not supported"),
         };
 
-        // Overwriting albedo_factor of CpuMesh in specified in the Asset3D
-        cpu_model.instances.iter().for_each(|instance| {
-            cpu_model.meshes[instance.mesh].materials[0].albedo_factor =
-                albedo_factor.map_or(re_renderer::Rgba::WHITE, |c| c.0.into());
-        });
+        // Overwriting albedo_factor of CpuMesh if specified in the Asset3D
+        if let Some(albedo_factor) = albedo_factor {
+            for instance in &cpu_model.instances {
+                cpu_model.meshes[instance.mesh].materials[0].albedo_factor = albedo_factor.0.into();
```

Comment:
I missed that in the previous PR as well, but why are we only changing the first material and not all materials? This should be looping over all materials of the mesh

Reply:
I though materials was a smallvec of size 1. Maybe I saw that in stl/obj and that may not be the case for gltf?
That is my bad :sweat: